src/Core/Grid/Query/CatalogPriceRuleQueryBuilder.php

```diff
@@ -203,8 +200,41 @@ private function applyFilters(QueryBuilder $qb, array $filters)
                 continue;
             }
 
+            if ($filterName === 'reduction') {
+                $numberOfDecimals = $this->findNumberOfDecimals($value);
+                // using TRUNCATE in order to have smart price searches
+                // Smart price searches means:
+                // searching for "10" will return both 10.0, 10.1 and 10.2
+                // searching for "10.0" will only return 10.0
+                // searching for "10.1" will only return 10.1
+                $qb->andWhere('TRUNCATE(' . $allowedFiltersAliasMap[$filterName] . ',' . $numberOfDecimals . ') = :' . $filterName);
+                $qb->setParameter($filterName, $value);
+                continue;
+            }
+
             $qb->andWhere($allowedFiltersAliasMap[$filterName] . ' LIKE :' . $filterName);
             $qb->setParameter($filterName, "%$value%");
         }
     }
+
+    /**
+     * @param string $value
+     *
+     * @return int
+     */
+    private function findNumberOfDecimals($value): int
+    {
+        if (!is_string($value)) {
+            throw new \InvalidArgumentException('Expected string');
+        }
+
+        $numberOfDecimals = 0;
+        $explodedValue = explode('.', $value);
+
+        if (count($explodedValue) > 1) {
```

Comment:
```suggestion
        if (isset($explodedValue[1])) {
```
Don't count the whole array if you only need the `[1]` index